hrms/payroll/doctype/salary_slip/salary_slip.py

```diff
@@ -950,7 +950,7 @@ def add_structure_components(self, component_type):
 			if self.salary_slip_based_on_timesheet and struct_row.salary_component == timesheet_component:
 				continue
 
-			amount = self.eval_condition_and_formula(struct_row, self.data)
+			amount = self.eval_condition_and_formula(struct_row, self.data) or 0
```

Comment:
This will affect condition checking at https://github.com/frappe/hrms/pull/551/files#diff-6809bf57e65fda9bc2f48a2da74b85c341ed2fce7245a360f42b4dca39e144beR977